source/modulo_core/src/communication/SubscriptionHandler.cpp

```diff
@@ -6,73 +6,103 @@ namespace modulo_core::communication {
 
 template<>
 SubscriptionHandler<std_msgs::msg::Bool>::SubscriptionHandler(std::shared_ptr<MessagePairInterface> message_pair) :
-    SubscriptionInterface(std::move(message_pair)) {}
+    SubscriptionInterface(std::move(message_pair)), clock_(std::make_shared<rclcpp::Clock>()) {}
```

Comment:
I am just seeing that now but is there a reason you are creating a new clock and not passing the one from the parent node at construction?

Reply:
It doesn't look like the clock object is too expensive to instantiate, and they refer to the same underlying time source. It might even be safer to have multiple clocks rather than one master instance, so that one clock doesn't have to deal with all the callbacks in the queue. However, if we change the clock type away from the default RCL_SYSTEM_TIME at any point, it would be difficult to propagate consistently, but that would only matter for time-critical behaviours. Throttled logging is not time-critical imo. 

TL;DR I think this is fine like this for now, and it makes the construction of the Handler in the component is less verbose